lib/contract_interact/pricer.js

```diff
@@ -0,0 +1,357 @@
+    ).then(function(transactionHash) {
+      return Promise.resolve(responseHelper.successWithData({transactionHash: transactionHash}));
+    });
+  },
+
+  /**
+  * Remove the price oracle address for a given currency
+  *
+  * @return {Promise}
+  *
+  */
+  unsetPriceOracle: function (senderAddress, senderPassphrase, currency, gasPrice) {
+    if (currency === undefined || currency === '') {
+      return Promise.reject('Currency is mandatory');
+    }
+    if (gasPrice === undefined || gasPrice === '' || gasPrice == 0 ) {
+      return Promise.reject('GasPrice is mandatory');
+    }
+    helper.assertAddress(senderAddress);
+
+    currContract.options.address = this.contractAddress;
+    currContract.setProvider( web3RpcProvider.currentProvider );
+    const transactionObject = currContract.methods.unsetPriceOracle(web3RpcProvider.utils.asciiToHex(currency));
+    const encodedABI = transactionObject.encodeABI();
+    return helper.sendTxAsyncFromAddr(
+      web3RpcProvider,
+      this.contractAddress,
+      encodedABI,
+      senderAddress,
+      senderPassphrase,
+      { gasPrice: gasPrice, gas: GAS_LIMIT }
+    ).then(function(transactionHash) {
+      return Promise.resolve(responseHelper.successWithData({transactionHash: transactionHash}));
+    });
+  },
+
+  /**
+  * Updates the acceptable margin range for a given currency
+  *
+  * @return {Promise}
+  *
+  */
+  setAcceptedMargin:  function (senderAddress, senderPassphrase, currency, acceptedMargin, gasPrice) {
+    if (currency === undefined || currency === '') {
+      return Promise.reject('Currency is mandatory');
+    }
+    if (gasPrice === undefined || gasPrice === '' || gasPrice == 0 ) {
+      return Promise.reject('GasPrice is mandatory');
+    }
+    if (acceptedMargin<0) {
+      return Promise.reject('Accepted margin cannot be negetive');
+    }
+    helper.assertAddress(senderAddress);
+
+    const transactionObject = currContract.methods.setAcceptedMargin(web3RpcProvider.utils.asciiToHex(currency), acceptedMargin);
+    const encodedABI = transactionObject.encodeABI();
+    return helper.sendTxAsyncFromAddr(
+      web3RpcProvider,
+      this.contractAddress,
+      encodedABI,
+      senderAddress,
+      senderPassphrase,
+      { gasPrice: gasPrice, gas: GAS_LIMIT }
+    ).then(function(transactionHash) {
+      return Promise.resolve(responseHelper.successWithData({transactionHash: transactionHash}));
+    });
+  },
+
+
+  /**
+  * Pay
+  *
+  * @return {Promise}
+  *
+  */
+  pay: function (
+    senderAddress,
+    senderPassphrase,
+    beneficiaryAddress,
+    transferAmount,
+    commissionBeneficiaryAddress,
+    commissionAmount,
+    currency,
+    intendedPricePoint,
+    gasPrice) {
+
+    if (gasPrice === undefined || gasPrice === '' || gasPrice == 0 ) {
+      return Promise.reject('GasPrice is mandatory');
+    }
+    if (transferAmount < 0) {
+      return Promise.reject('Transfer amount cannot be negetive');
```

Comment:
in validations instead of Promise.reject let's use responseHelper.error method for consistent response format

Reply:
Done